src/app/getting-started/services/provider.service.ts

```diff
@@ -31,8 +36,42 @@ export class ProviderService {
    *
    * @param redirect URL to be redirected to after successful account linking
    */
-  linkGitHub(redirect: string): void {
-    this.link("github", redirect);
+  linkGitHub(redirect: string): void{
+    let headers = new Headers();
+    let githubLinkURL = this.linkUrl + `redirect=` + redirect ;
+
+    this.http
+    .get(this.linkUrl, { headers: headers })
+    .map(response => {
+      let redirectInfo = response.json() as Link;
+      this.redirectToAuth(redirectInfo.redirect);
+    })
+    .catch((error) => {
+      return this.handleError(error);
+    });
+  }
+
+   /**
+   * Link a GitHub and OpenShift Online account to the user account
+   *
+   * @param redirect URL to be redirected to after successful account linking
+   */
+  linkGitHubAndOpenShift(redirect: string): void {
+
+    let parsedToken = jwt_decode(this.auth.getToken());
+    // the new url is /api/link/session 
+    let opensShiftLinkingRedirectUrl = this.getLegacyLinkingUrl("openshift-v3",redirect);
+    this.linkGitHub(opensShiftLinkingRedirectUrl);
```

Comment:
This doesn't really make sense to me - you are asking for the openshift url and then calling linkGitHub.  What should the logic be?

Reply:
When we link Github + OpenShift together:
We call the new linking API for github and pass the old linking api url for OpenShift in the redirect.

Since we are yet to move over openshift-v3 while we wait for OSO proxy to be in place.